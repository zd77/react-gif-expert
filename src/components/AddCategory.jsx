import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
    const [ inputValue, setInputValue ] = useState('')
    const onInputChanged = ( event ) => {
        setInputValue( event.target.value )
    }
    const onSubmit = ( event ) => {
        event.preventDefault()
        if( inputValue.trim().length <= 1 ) return
        onNewCategory( inputValue.trim() )
        // setCategories(( categories ) => [ ...categories, inputValue ])
        setInputValue('')
    }
    return (
        <form onSubmit={ onSubmit } aria-label='form' >
            <input 
                type='text'
                placeholder='Search for GIFs'
                value={ inputValue }
                onChange={ onInputChanged }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired    
}
