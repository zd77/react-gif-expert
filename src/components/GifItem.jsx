import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {
    return (
        <div className='card' >
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
// Validando existencia y tipo de las propiedas necesarias
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}