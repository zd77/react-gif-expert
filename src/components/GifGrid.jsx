import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs( category )

    return (
        <Fragment>
            <h3>{ category }</h3>
            {
                isLoading
                ? <h2>Cargando...</h2>
                : null
            }
            <div className='card-grid' >
                {
                    images.map( image => (
                        <GifItem 
                            key={ image.id } 
                            { ...image }
                        />
                    ))
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}