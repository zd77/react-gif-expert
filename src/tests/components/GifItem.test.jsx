import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem'

describe('Tests on <GifItem />', () => {
    const title = 'Title'
    const url = 'http://localhost/Url'
    test('should match with the snapshot', () => { 
        const { container } = render( <GifItem title={ title } url={ url }/> )
        expect( container ).toMatchSnapshot();
     })

     test('should show the img with the indicated URL $ ALT', () => { 
        render( <GifItem title={ title } url={ url }/> )
        // screen.debug()
        // expect( screen.getByRole('img').src ).toBe( url )
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
      })

      test('should show the title in the component <p></p>', () => { 
        render( <GifItem title={ title } url={ url }/> )
        expect( screen.getByText( title) ).toBeTruthy()
       })
})