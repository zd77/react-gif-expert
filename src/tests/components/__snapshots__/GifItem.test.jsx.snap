// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="http://localhost/Url"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
